sql/src/main/java/org/apache/druid/sql/SqlExecutionReporter.java

```diff
@@ -94,6 +94,9 @@ public void emit()
         metricBuilder.setDimension("id", plannerContext.getSqlQueryId());
         metricBuilder.setDimension("nativeQueryIds", plannerContext.getNativeQueryIds().toString());
       }
+      if (stmt.sqlToolbox.engine != null) {
+        metricBuilder.setDimension("engine", stmt.sqlToolbox.engine.name());
```

Comment:
What is the correct way to get a reference to the engine? I also noticed it's available via the plannerContext.

Reply:
heh, the statement also has a plannerContext, so could do `plannerContext.getEngine().name()` or `stmt.plannerContext.getEngine().name()` or what you did here... probably others too 😜 

Most of the other stuff looks like it is coming from `plannerContext` so that seems probably best?

Reply:
I chose this instead of `plannerContext` because it looks like `sqlToolbox` is never null, while `plannerContext` might be null. Also, having to a nested if to check that the engine in the plannerContext is not null, seemed like it would make the code harder to read / follow.

Reply:
The toolbox is never null. The engine is also never null, so no need for the `null` check.

Reply:
@paul-rogers The sqlEngine in the SqlToolbox provided by the SqlStatementFactoryModule is null

https://github.com/apache/druid/blob/master/sql/src/main/java/org/apache/druid/sql/guice/SqlModule.java#L173